### ISSUE TITLE
Trade API white list

### ISSUE THREAD

**Issue by gdneye** [licoffe/POE-sniper#82]
Hey Licoffe,

Does this

https://www.pathofexile.com/forum/view-thread/2036957

essentially kill your tool?  To me, a 60 second delay would completely eliminate the purpose of using POE-sniper.  I'll probably test it out at the beginning of the league, but my expectation is that I will be using poe.trade, poeapp and pathofexile.com/trade for the foreseeable future.

Assuming this kills your app, thank you for all of the work you've put in, I've greatly appreciated it.

Gdneye

**Comment by LordFrz** [comment]
Might be a good backup if poe.trade goes down. Poe.trade is always super laggy for me, so even at a 60sec delay, this tool should still be usable for normal purchases atleast.

**Comment by ghost** [comment]
I don't think it kills the app. It's still pretty useful to snipe expansive items like starforge

**Comment by gdneye** [comment]
@Morgoth92 what do you mean?  Wouldn't you still be better off using poe.trade (or similar) to snipe a starforge?

@licoffe Would you consider making a public website (that could be whitelisted) and/or partnering with poe.trade or (more likely) poeapp?  Poeapp seems genuinely interested in making a solid trading website, but is honestly lacking in UI skills.  

Having a tool that responded more quickly than poe.trade was nice, but the big QoL features of this tool were the UI.  Having 100% of my searches in one window (instead of spread out over 60 internet tabs) was an enormous benefit especially considering I also play with PoB, a build guide, the PoE wiki, youtube for music, poe.ninja, poe-affix, etc etc open.  Being able to bucket those searches was great, the results being in one location combined with the auto-copy to clipboard meant that anytime I hear a whoop I don't have to search through 60 windows (that have to be stacked in a specific way so I can see and get to them all), I just ctrl v in game.  On top of that the ability to customize things (such as league, bo only, conversion, copy to clipboard, etc etc) on the fly for individual filters or an entire group.....

I could go on.  But in a few months this app has blown away poe.trade and poeapp.  Poe.trade barely seems to try and while poeapp seems like a really nice guy who genuinely wants to make a great site, stay on top of new mods, add new features, etc, his UI skills are lacking and the inability to leave his site running while you are afk is kind of a pain (although I would take that trade-off if I could reset all 60 filters at once rather than needing to click through 60 different tabs).

Anyways... I'm (obviously) a little bit frustrated by the update.  I've been loving this app.  If you either turned it into a website or partnered with poeapp to improve his UI I would be ecstatic.  I understand that the major issue with running a website would be the server fees.  I, personally, would be glad to throw some money into a patronage (or possibly contribute more as an investment if it was realistic that ad revenue/patronage would eventually make that investment worthwhile) for that to happen and  to assist in sparking interest in the community.  This app is soo far beyond the other tools available that I think people would show a lot of interest in it and transition from poe.trade and poeapp quickly.

I hope you do something that allows this app to be whitelisted, because I don't really need to be ahead of the other guys, I just want an interface that works.  If not, thank you for the tremendous job you've done up to this point.  I've sincerely appreciated it.

**Comment by gdneye** [comment]
Another idea... How difficult would it be to allow poeapp, poe.trade or pathofexile/trade to run the searches, but use your UI?  And how much delay would doing so create?  That could be a nice solution that would prevent you from taking on server costs, while still maintaining the nice UI.  Ideally it could have the ability to switch which of the 3 trade sites it was connected to based on which one was responding the most quickly at that time.

**Comment by ghost** [comment]
I agree about the project ideas of #gdneye

**Comment by licoffe** [comment]
Thank you everyone for the kind words, it's very appreciated! I've sent a mail to the GGG support to see if they would consider whitelisting poe-rates.com, the backend where the chunks are pulled from. Chris replied that he would get back to me on Monday since he needs to discuss it with other people from the team.

**Comment by gdneye** [comment]
Licoffe,

Have you considered converting the app to a website (that would also use poe-rates as the backend)?  I mention this because this reddit thread

https://www.reddit.com/r/pathofexile/comments/7j5cet/trade_api_delays/

includes this post

_I think this is why the proposition of "What if the tool has to be run on your computer?" is flawed from the beginning. It would be an interesting discussion to have, if not for the fact that all of this happened because GGG didn't want everybody to run something locally due to bandwidth limits._

I recognize that using a desktop app and pulling from a single backend has the same effect on GGG as using a website, but I think that this guy (and the people who upvoted him) is accidentally conflating "locally ran" with "locally pulling the data".  I imagine that any concerns over a desktop app will be presented by individuals with a fundamental misunderstanding of how these apps work, but...

There is also this comment

_Should not be executables. I'm concerned about the reliance on downloaded content in PoE anyway, and the message it sends to players. Many players are using AHK, PoB, and loot filters already, and there are real security concerns here. Online tools limit this exposure._


I think that the GGG community will be naturally distrustful of any desktop app.  Since GGG is asking the community for feedback on what the rules should be, it seems the likelihood of a web app being approved is higher.  (It's possible you've already heard from GGG on if you will be whitelisted in which case this may be a moot point.)

Would there be any disadvantages to running this as a web app?  Would a web app be slower?  Would it be difficult to migrate the UI and/or keep it the same?  Would it be difficult to save all searches in one tab, organize folders for that tab, and save the info on a web app?  Or lastly, are you simply more experienced with building a desktop app?

**Comment by gdneye** [comment]
PS I know you are probably just enjoying the first week of the league.  I absolutely do not expect you to drop your league play to resolve this.  I'm just mentioning these things now because I am thinking of them now.

**Comment by pedrolucasmag** [comment]
It just pull data from poe.trade and poe.trade is white listed so there is no problem

**Comment by licoffe** [comment]
> Have you considered converting the app to a website (that would also use poe-rates as the backend)?

Even though there are no clear guidelines yet (Chris said the matter is still actively discussed within the team after the reddit/poe posts), I also suspect that only websites will have the right to be whitelisted and I started writing a tool on the backend side to filter items.

> Would there be any disadvantages to running this as a web app? Would a web app be slower? Would it be difficult to migrate the UI and/or keep it the same? Would it be difficult to save all searches in one tab, organize folders for that tab, and save the info on a web app? Or lastly, are you simply more experienced with building a desktop app?

The incentive behind the current tool were the following:
- People download the raw API data, at their own speed, provided AS IS by GG, i.e. no tempering on the data from any party
- Filtering on the backend would probably require a lot of processing power, let's harness the CPU power of everyone. Somebody wants to use 500 filters, the search is running on his hardware so there's no issue

With a website however, you won't have the nice notifications I added in the last versions and I will probably have to impose a limit on the amount of filters sent by IP. Other than that, I don't see any particular issue.

The envisioned workflow is the following:
- People create JSON filter descriptions using either the website or the client
- The filters are sent to the backend and read by the live search algorithm
- Matching results are saved and sent back to either the website or the client

I started working on a rewrite of the matching algorithm, this time in C++ to achieve better performances. I got basic matching done (price, properties) as well as most of the workflow (data transmission). I will soon start to work on integrating this approach in the current client.
